### ISSUE TITLE
Question regarding reproduction results on HumanML3D and KIT-ML

### ISSUE THREAD

**Issue by pidanoyes** [seokhyeonhong/salad#15]
## Question regarding reproduction results on HumanML3D and KIT-ML

Hi authors,

First of all, thank you so much for your excellent work and for open-sourcing the code for SALAD! It is a great contribution to the community.

I am currently trying to reproduce the results reported in your paper on both the HumanML3D and KIT-ML datasets. However, I noticed some performance gaps between my reproduced results and the official numbers reported in the paper. 

Specifically, **the FID metric on the KIT-ML dataset shows a significant gap**, and the **R-Precision (Top-1, Top-2, Top-3) metrics on both datasets consistently fall short** of the paper's results.

Here is the detailed comparison of the evaluation metrics:

### 1. HumanML3D Results Comparison

| Metric | Reproduced Results | Official Paper Results |
| :--- | :--- | :--- |
| **Top-1** | 0.570 (conf. 0.003) | **0.581** |
| **Top-2** | 0.761 (conf. 0.003) | **0.769** |
| **Top-3** | 0.850 (conf. 0.002) | **0.857** |
| **FID** | 0.091 (conf. 0.003) | **0.076** |
| **Matching** | 2.695 (conf. 0.009) | 2.649 |
| **Diversity** | 9.726 (conf. 0.103) | 9.696 |
| **MultiModality** | 1.812 (conf. 0.069) | 1.751 |

### 2. KIT-ML Results Comparison

| Metric | Reproduced Results | Official Paper Results |
| :--- | :--- | :--- |
| **Top-1** | 0.464 (conf. 0.007) | **0.477** |
| **Top-2** | 0.696 (conf. 0.008) | **0.711** |
| **Top-3** | 0.819 (conf. 0.005) | **0.828** |
| **FID** | **0.441** (conf. 0.024) | **0.296** |
| **Matching** | 2.635 (conf. 0.017) | 2.585 |
| **Diversity** | 11.206 (conf. 0.092) | 11.097 |
| **MultiModality** | 0.986 (conf. 0.046) | 1.004 |

### 3. Training and Testing Commands Used

**For HumanML3D and KIT-ML:**
```bash
# VAE (for HumanML3D)
python train_vae.py --name vae_human --dataset_name t2m --gpu_id 0 --batch_size 64
# Denoiser (for HumanML3D)
python train_denoiser.py --name denoiser_human --vae_name vae_human --gpu_id 0 --batch_size 64
# Test (for HumanML3D)
python test_vae.py --name vae_human --gpu_id 0
python test_denoiser.py --name denoiser_human --num_inference_timesteps 50 --gpu_id 0

# VAE (for KIT-ML)
python train_vae.py --name vae_kit --dataset_name kit --gpu_id 2 --batch_size 16
# Denoiser (for KIT-ML)
python train_denoiser.py --name denoiser_kit --vae_name vae_kit --gpu_id 2 --dataset_name kit --batch_size 16
# Test (for KIT-ML)
python test_vae.py --name vae_kit --gpu_id 2 --dataset_name kit
python test_denoiser.py --name denoiser_kit --num_inference_timesteps 50 --gpu_id 2 --dataset_name kit
```

[denoiser_human_opt.txt](https://github.com/user-attachments/files/26004137/denoiser_human_opt.txt)

[denoiser_kit_opt.txt](https://github.com/user-attachments/files/26004138/denoiser_kit_opt.txt)

[vae_kit_opt.txt](https://github.com/user-attachments/files/26004155/vae_kit_opt.txt)

[vae_human_opt.txt](https://github.com/user-attachments/files/26004156/vae_human_opt.txt)

For your convenience, I have also attached the corresponding configuration files (vae_human_opt.txt, denoiser_human_opt.txt, vae_kit_opt.txt, denoiser_kit_opt.txt) to this issue.

Could you kindly provide some insights into what might be causing these discrepancies? Are there any specific hyperparameter settings (e.g., learning rate, training epochs), random seeds, or evaluation tricks that I might have missed?

Please let me know if you need any further information (e.g., training logs) from my side.

Thanks again for your time and help!

Best regards,
Arthur

**Comment by JHang2020** [comment]
The same question...
